joomla/joomla-cms
# Missed few mootools modals on com_menus boostrapization

#### This is complimentary to #4661
- Some modals for selecting a single article, contact and newsfeed are still based on mootools code
- Also a disabled="disabled" was added to the menu type input and that prohibits validation to act properly, thus is now removed
- The validation for the modal field wasn’t very descriptive
- Editing a module in `Module Assignment` and then canceling the changes will result in a lock icon appearing in module manager
- All modals have a footer with close button and modules modal also have a save & close button as well.
#### testing
1. Create a menu, add some title and alias and press save you should see:
   ![screen shot 2015-04-19 at 5 26 40](https://cloud.githubusercontent.com/assets/3889375/7219788/5420062c-e6b9-11e4-91db-6fdb50cfad36.png)
2. On the home page menu open module assignment click on a module to bring the modal and the click cancel. Go to module manager and verify that no lock icon exist in front of the module you edit in homepage menu.
3. Create a new menu and select as Menu Item Type  `singe article` then click on select button on the next field `Select Article` respectively you should see a bootstrap modal:
   ![screen shot 2015-04-19 at 5 39 36](https://cloud.githubusercontent.com/assets/3889375/7219822/1e6b87fc-e6bb-11e4-9396-5f523a91ab7a.png)
   Repeat this also for `single contact` `single newsfeed` with a selection on `Select Contact` `Select Newsfeed` respectively.
4. Also Repeat the step 2 insert a title but DON’T select an article, newsfeed and contact and press save you should see a descriptive message about the field that needs input (previously you would only see an error as a message!) Thanks @Erftralle for coding this!
#### Preview

![screen shot 2015-04-29 at 2 15 17](https://cloud.githubusercontent.com/assets/3889375/7382379/d9636fae-ee16-11e4-99fe-d61c2004368e.png)
![screen shot 2015-04-29 at 2 15 40](https://cloud.githubusercontent.com/assets/3889375/7382381/dd0ff3e8-ee16-11e4-8836-0f3c6cdf8a86.png)
![screen shot 2015-04-29 at 2 15 57](https://cloud.githubusercontent.com/assets/3889375/7382384/e16fe858-ee16-11e4-88b6-9bf53c83ccab.png)
![screen shot 2015-04-29 at 2 17 01](https://cloud.githubusercontent.com/assets/3889375/7382386/e5f3926c-ee16-11e4-8d62-d6e22fb4958c.png)
![screen shot 2015-04-29 at 2 17 32](https://cloud.githubusercontent.com/assets/3889375/7382388/e9588796-ee16-11e4-8eec-79248b43bac1.png)
